Applications/Spire/Source/Ui/ScrollBar.cpp

```diff
@@ -0,0 +1,123 @@
+#include "Spire/Ui/ScrollBar.hpp"
+#include <QApplication>
+#include <QHBoxLayout>
+#include <QMetaEnum>
+#include "Spire/Ui/Box.hpp"
+#include "Spire/Spire/Dimensions.hpp"
+
+using namespace boost;
+using namespace boost::signals2;
+using namespace Spire;
+using namespace Spire::Styles;
+
+namespace {
+  const auto SCROLL_BAR_SIZE = 13;
+
+  auto DEFAULT_STYLE() {
+    auto style = StyleSheet();
+    style.get(Any()).
+      set(BackgroundColor(QColor::fromRgb(0xFF, 0xFF, 0xFF)));
+    style.get(ScrollBarThumb()).
+      set(BackgroundColor(QColor::fromRgb(0xC8, 0xC8, 0xC8)));
+    return style;
+  }
+}
+
+ScrollBar::ScrollBar(Qt::Orientation orientation, QWidget* parent)
+    : StyledWidget(parent) {
+  auto layout = new QHBoxLayout(this);
+  layout->setContentsMargins({});
+  m_scroll_bar = new QScrollBar(orientation, this);
+  m_scroll_bar->setContextMenuPolicy(Qt::NoContextMenu);
+  m_scroll_bar->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
+  layout->addWidget(m_scroll_bar);
+  set_style(DEFAULT_STYLE());
+  connect(m_scroll_bar, &QScrollBar::valueChanged, [=] (auto value) {
+    m_position_signal(value);
+  });
+}
+
+Qt::Orientation ScrollBar::get_orientation() {
+  return m_scroll_bar->orientation();
+}
+
+ScrollBar::Range ScrollBar::get_range() const {
+  return {m_scroll_bar->minimum(), m_scroll_bar->maximum()};
+}
+
+void ScrollBar::set_range(const Range& range) {
+  m_scroll_bar->setRange(range.m_start, range.m_end);
+}
+
+void ScrollBar::set_range(int start, int end) {
+  m_scroll_bar->setRange(start, end);
+}
+
+int ScrollBar::get_line_size() const {
+  return QApplication::wheelScrollLines();
```

Comment:
The line size is not the same as the wheel scroll. The line size is the minimum scrollable unit and the wheel scroll is some constant multiple of that unit where the constant is usually 3.

Reply:
I know that, so I asked you this question on the Asana. The line size is actually not used in our scroll bars, and the minimum unit for wheel scroll is needed in the specs. The line size can be used to calculate the unit of the wheel scroll, but it's not a straightforward way to set the unit of the wheel scroll.

Reply:
```get_line_size``` will need to return the line size, not the wheel scroll size.

Reply:
I see.